src/main/java/taxi/dao/jdbc/CarJdbcDao.java

```diff
@@ -0,0 +1,199 @@
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant get car by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversFromCar(car.getId()));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        String getAllCars = "SELECT * FROM cars WHERE deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(getAllCars)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResult(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create list of cars", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversFromCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(updateQuery)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't update car " + car.getId(), e);
+        }
+        deleteDriversFromCars(car);
+        addDriversToCar(car);
+        return car;
+        // remove old relations between this car(car id) and drivers(from driver_cars table)
+        // delete from cars drivers where car id = carid (from update) -> insert for each
+        // insert new relations using car.getDrivers (71 line) for each driver adding new cars
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars SET deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(deleteQuery)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't delete car - id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriverId(Long driverId) {
+        String getAllCarsByDriver = "Select * FROM driver_cars dc"
+                + " INNER JOIN cars c ON dc.driver_id = c.id"
+                + " WHERE dc.driver_id = ? AND c.deleted = false ";
```

Comment:
we need the info about the manufacturer for `getCarFromResult` method
also we need to join drivers table to check that driver is not deleted

we should have 3 joins here